base/MAPL_GranuleGridFactory.F90

```diff
@@ -0,0 +1,1257 @@
+#include "MAPL_Exceptions.h"
+#include "MAPL_ErrLog.h"
+#include "unused_dummy.H"
+
+! This module generates ESMF_Grids for logically-rectangular grid based on observation (netCDF)
+! Spacing between lats (lons) needs not be constant (imhomogenous)
+! LatLon concepts such as corners, pole and dataline are eliminated
+
+module MAPL_GranuleGridFactoryMod
+   use MAPL_AbstractGridFactoryMod
+   use MAPL_MinMaxMod
+   use MAPL_KeywordEnforcerMod
+   use MAPL_ExceptionHandling
+   use MAPL_ShmemMod
+   use mapl_ErrorHandlingMod
+   use MAPL_Constants
+   use ESMF
+   use pFIO
+   use MAPL_CommsMod
+   use, intrinsic :: iso_fortran_env, only: REAL32
+   use, intrinsic :: iso_fortran_env, only: REAL64
+   implicit none
+   private
+
+   public :: GranuleGridFactory
+
+   integer, parameter :: NUM_DIM = 2
+   
+   type, extends(AbstractGridFactory) :: GranuleGridFactory
+      private
+      character(len=:), allocatable :: grid_name
+      character(len=:), allocatable :: grid_file_name      
+      ! Grid dimensions
+      integer :: im_world = MAPL_UNDEFINED_INTEGER
+      integer :: jm_world = MAPL_UNDEFINED_INTEGER
+      integer :: lm = MAPL_UNDEFINED_INTEGER
+      real(kind=REAL64), allocatable :: lon_centers(:)
+      real(kind=REAL64), allocatable :: lat_centers(:)
```

Comment:
@bena-nasa  I think you are referring to this array, not used in the following code.  I see it is mainly used in  `subroutine initialize_from_file_metadata`, which is not used as initilization.  And this initialization array is
has a 1D dimension, which is the starting address in Fortran array.    It appears that in swath we need lon(nx, ny).  But I am not very sure.  But the bottom line is to make consistency as in `file_metadata` related procedures.